lsp/nls/src/requests/completion.rs

```diff
@@ -21,42 +203,45 @@ pub fn handle_completion(
         .id_of(params.text_document_position.text_document.uri.as_str())
         .unwrap();
 
+    let text = server.cache.files().source(file_id);
     let start = position_to_byte_index(
         server.cache.files(),
         file_id,
         &params.text_document_position.position,
     )
     .unwrap();
 
-    let locator = (file_id, ByteIndex(start as u32));
+    // -1 because at the current cursor position, there is no linearization item there,
+    // so we try to get the item just before the cursor.
+    let locator = (file_id, ByteIndex((start - 1) as u32));
```

Comment:
Nitpick: I guess there may be an item at the cursor (for example partial identifiers are valid identifiers). I think you rather want this for record completion to work, and it turns out it's fine for normal completion too, because if you started typing an identifier, the position one byte before is the same token as well?

Reply:
Yes. It works well with normal completion. This is just for record completion to work